libraries/chain/include/eosio/chain/transaction_context.hpp

```diff
@@ -83,12 +99,14 @@ namespace eosio { namespace chain {
 
          void validate_referenced_accounts( const transaction& trx, bool enforce_actor_whitelist_blacklist )const;
 
+         uint32_t get_action_id() const { return action_id.current(); }
+
       private:
 
          friend struct controller_impl;
          friend class apply_context;
 
-         void add_ram_usage( account_name account, int64_t ram_delta );
+         void add_ram_usage( account_name account, int64_t ram_delta, const ram_trace&& ram_trace );
```

Comment:
Should be `const ram_trace&`. And again consider renaming to avoid same name for type and argument.

Reply:
I've changed all occurrences to `const ram_trace& trace`. Not that much better but are least, there are different.